beam/client.go

```diff
@@ -0,0 +1,57 @@
+package beam
+
+import (
+	"bytes"
+	"errors"
+	"time"
+
+	"github.com/RTradeLtd/rtfs"
+)
+
+// Laser is used to transfer content between two different private networks
+type Laser struct {
+	net1 *rtfs.IpfsManager
+	net2 *rtfs.IpfsManager
+}
+
+// NewLaser creates a laser client to beam content between different ipfs networks
+func NewLaser(net1URL, net2URL string) (*Laser, error) {
+	net1, err := rtfs.NewManager(net1URL, nil, time.Minute*10)
+	if err != nil {
+		return nil, err
+	}
+	net2, err := rtfs.NewManager(net2URL, nil, time.Minute*10)
+	if err != nil {
+		return nil, err
+	}
+	return &Laser{
+		net1: net1,
+		net2: net2,
+	}, nil
+}
+
+// Beam is used to transfer content bewween two different networks
+func (l *Laser) Beam(sourceNet int, contentHash string) error {
```

Comment:
if you use the `source`, `dest` you can use a `bool` here

thought I'm not sure when a `Laser` will need to transfer data both ways? from the other PR it looks like this is constructed on a per-request basis and does one transaction

Reply:
I want to stay away from explicitly labelling a network source or destination, since let's say a user wants to transfer 2 pieces of content during an API call to Temporal, one from net1->net2, and another from net2->net1, by labelling them "source" or "destination" we would need to then instantiate 2 instances of `Laser` as opposed to keeping with net1 and net2 we would only need to instantiate 1 instance of `Laser`

Reply:
Hm, I think the `1` and `2` paradigm might be even more confusing, since it's less clear which network is which. For back and forth, you might want to expose an API like:

```go
type Beamer interface {
  BeamTo(hash string) error
  BeamFrom(hash string) error
}
```